### ISSUE TITLE
Error: Skipping incompatible libraries

### ISSUE THREAD

**Issue by ovidiugabriel** [crystal-lang/crystal#5738]
I installed Crystal on  CentOS 7 (Linux 3.10.0-327.36.1.el7.x86_64) using the instructions at
https://crystal-lang.org/docs/installation/on_redhat_and_centos.html

Crystal 0.24.1 (2017-12-22)
LLVM: 4.0.0
Default target: x86_64-unknown-linux-gnu

While I am running it for a simple hello world application I get the following errors:

/usr/bin/ld: skipping incompatible /usr/lib/libpthread.so when searching for -lpthread
/usr/bin/ld: skipping incompatible /usr/lib/librt.so when searching for -lrt
/usr/bin/ld: skipping incompatible /usr/lib/libdl.so when searching for -ldl
/usr/bin/ld: skipping incompatible /usr/lib/libc.so when searching for -lc

**Comment by RX14** [comment]
These look like warnings to me, could you paste the entire program output?

**Comment by ovidiugabriel** [comment]
Yes, it is true that the binary artifact is generated if I am using `crystal build`. So you are right it is just a warning.  If the program has any output and I run it I am able to see the output.

Any known reason for this? the default target of Crystal is aligned with my machine architecture, I don't see any reason for incompatibility.

**Comment by RX14** [comment]
Seems likely that your system is still using `/usr/lib` for 32bit libraries and `/usr/lib64` for 64bit libraries (most distros don't do this any more, you seem to be using a super old distro which also explains #5739). Crystal hardcodes `/usr/lib` as a library path (which it shouldn't, and should be removed) which means the linker prints a warning as it skips the 32bit libraries in `/usr/lib`. The linking succeeds correctly using the libraries in `/usr/lib64`.

**Comment by kirbyfan64** [comment]
Actually I think OP is using CentOS 7.2, which was releases in 2015. It seems that CentOS isn't going to be changing the directory layout for the 7.x series at all, including the latest 7.4 (from 2017).

**Comment by ZaWertun** [comment]
I see same errors on Fedora 28:
```
$ crystal play
Listening on http://127.0.0.1:8080
/usr/bin/ld: skipping incompatible /usr/lib/libm.so when searching for -lm
/usr/bin/ld: skipping incompatible /usr/lib/libpthread.so when searching for -lpthread
/usr/bin/ld: skipping incompatible /usr/lib/librt.so when searching for -lrt
/usr/bin/ld: skipping incompatible /usr/lib/libdl.so when searching for -ldl
/usr/bin/ld: skipping incompatible /usr/lib/libc.so when searching for -lc
```

**Comment by jhass** [comment]
Since these are just warnings, is there anything we can do here?
